src/ucp/wireup/ep_match.c

```diff
@@ -102,8 +102,8 @@ static void ucp_ep_match_insert_common(ucp_ep_match_ctx_t *match_ctx,
 {
     ucs_assert(!(ep->flags & UCP_EP_FLAG_ON_HASH));
     ucp_ep_match_list_add_tail(list, &ucp_ep_ext_gen(ep)->ep_match.list);
-    ep->flags                     |= UCP_EP_FLAG_ON_HASH;
-    ucp_ep_ext_gen(ep)->dest_uuid  = dest_uuid;
+    ep->flags                              |= UCP_EP_FLAG_ON_HASH;
```

Comment:
Maybe rename this flag to smt like `UCP_EP_FLAG_ON_MATCH_CTX`? we don't have direct worker level hash storage anymore. It's encapsulated inside matching context.